### ISSUE TITLE
There is no Windows version to self-host ntfy

### ISSUE THREAD

**Issue by JPMakesStuff** [binwiederhier/ntfy#1104]
It would seem the ntfy windows version is only for displaying notifications and not acting as a self-hosted server ???

Could a Windows server version which runs on a custom TCP port be released?

This would be very desirable for home automation setups where the *less-than-stellar* vendor only supports their own notification method but it's very slow or down most of the time.... Some example applications that run on Windows but notifications arrive too late or not at all: HomeSeer, NxWitness, Sighthound video, SyncBackSE

Yeah, I *could* run it in Docker, but ... why?

**Comment by user8446** [comment]
Why not run it in WSL2? I run a Nextcloud server in WSL2 and it's just as stable as a standalone server.

**Comment by kilanny** [comment]
Is it possible to cross build it from linux to target windows?
Some people have a VPS server of windows server. They cannot run WSL2, hyper-v or even vmware/virtual box.

**Comment by kilanny** [comment]
I have successfully compiled in ubuntu for windows (just server without docs or web page). Removed a few things in server.go that were causing errors, and done

```
sudo apt-get install gcc-multilib

sudo apt-get install gcc-mingw-w64

GOOS=windows GOARCH=amd64 CGO_ENABLED=1 CXX=x86_64-w64-mingw32-g++ CC=x86_64-w64-mingw32-gcc /usr/local/go/bin/go build
```

**Comment by JPElectron** [comment]
Yeah, but then I'm having to run WSL also...  I guess I'll take the hint dev dislikes Windows and I should go elsewhere, like [Gotify](https://gotify.net/) or other suggestion?

**Comment by wtf911** [comment]
[ntfy.zip](https://github.com/user-attachments/files/20031008/ntfy.zip)

For testing purposes here is the latest commit compiled for Windows.

**Comment by sayyyed** [comment]
> [ntfy.zip](https://github.com/user-attachments/files/20031008/ntfy.zip)
> 
> For testing purposes here is the latest commit compiled for Windows with server working.

thanks, i tried it and works very well, i am using it in production

**Comment by binwiederhier** [comment]
Thanks for testing. I will review again what it takes to make the Windows version. Looks like it's already working without changes.

**Comment by binwiederhier** [comment]
@wtf911 Is this still the only changes you had to make? https://github.com/binwiederhier/ntfy/pull/1328/changes

**Comment by wtf911** [comment]
@binwiederhier Yes, that's correct. That's all I did to be able to compile the Windows version and I added to be able to run it as a Windows service. Happy to see this getting some attention. :)

**Comment by binwiederhier** [comment]
Done https://github.com/binwiederhier/ntfy/pull/1552 -- The next server release will contain a Windows server version. Tested on Win 11. Service is working too.
